core/src/main/java/net/sourceforge/jnlp/cache/cache/Cache.java

```diff
@@ -0,0 +1,173 @@
+import net.sourceforge.jnlp.cache.CacheUtil;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.List;
+
+/**
+ * The Cache
+ */
+public class Cache {
+    /**
+     * Clears the cache by deleting all content
+     * <p>
+     * Note: Because of how our caching system works, deleting jars of another javaws
+     * process is using them can be quite disastrous. Hence why Launcher creates lock files
+     * and we check for those.
+     *
+     * @return true if the cache could and was cleared
+     */
+    public static boolean clearCache() {
+        return CacheLRUWrapper.getInstance().clearCache();
+    }
+
+    /**
+     * This will remove all old cache items.
+     */
+    public static void cleanCache() {
+        CacheLRUWrapper.getInstance().cleanCache();
+    }
+
+    /**
+     * Returns the file for the locally cached contents of the
+     * resource.  This method returns the file location only and does
+     * not download the resource.  The latest version of the
+     * resource that matches the specified version will be returned.
+     *
+     * @param resource the resource {@link URL}
+     * @param version  the version of the local file
+     * @return the file location in the cache, or {@code null} if no versions cached
+     * @throws IllegalArgumentException if the resource is not cacheable
+     */
+    public static File getCacheFile(final URL resource, final VersionId version) {
+        if (!CacheUtil.isCacheable(resource)) {
+            throw new IllegalArgumentException(resource + " is not a cacheable resource");
+        }
+        return CacheLRUWrapper.getInstance().getOrCreateCacheFile(resource, version);
+    }
+
+    public static void addToCache(ResourceInfo infoFromRemote, InputStream unpackedStream) throws IOException {
+        if (!CacheUtil.isCacheable(infoFromRemote.getResourceHref())) {
+            throw new IllegalArgumentException(infoFromRemote.getResourceHref() + " is not a cacheable resource");
+        }
+        CacheLRUWrapper.getInstance().addToCache(infoFromRemote, unpackedStream);
+    }
+
+    /**
+     * Invalidate the entry and make it eligible for removal.
+     *
+     * @param resource the resource {@link URL}
+     * @param version  the versions
+     * @return the newly created cache file
+     * @throws IllegalArgumentException if the resource is not cacheable
+     */
+    public static File replaceExistingCacheFile(final URL resource, final VersionId version) {
+        if (!CacheUtil.isCacheable(resource)) {
+            throw new IllegalArgumentException(resource + " is not a cacheable resource");
+        }
+        return CacheLRUWrapper.getInstance().replaceExistingCacheFile(resource, version);
+    }
+
+    public static void deleteFromCache(ResourceInfo info) {
+        CacheLRUWrapper.getInstance().deleteFromCache(info.getResourceHref(), info.getVersion());
```

Comment:
null check / assert for info

Reply:
internal use with non null arguments only